subprojects/ear/src/main/java/org/gradle/plugins/ear/EarPlugin.java

```diff
@@ -99,20 +100,25 @@ public FileCollection call() throws Exception {
                         if (project.getPlugins().hasPlugin(JavaPlugin.class)) {
                             return null;
                         } else {
-                            return project.fileTree(earPluginConvention.getAppDirName());
+                            return project.fileTree(task.getAppDirName());
                         }
                     }
                 });
             }
         });
     }
 
-    private void configureWithJavaPluginApplied(final Project project, final EarPluginConvention earPluginConvention, PluginContainer plugins) {
+    private void configureWithJavaPluginApplied(final Project project, final TaskProvider earTask, PluginContainer plugins) {
         plugins.withType(JavaPlugin.class, javaPlugin -> {
             final JavaPluginExtension javaPluginExtension = project.getExtensions().findByType(JavaPluginExtension.class);
 
             SourceSet sourceSet = mainSourceSetOf(javaPluginExtension);
-            sourceSet.getResources().srcDir((Callable) () -> earPluginConvention.getAppDirName());
+            project.getTasks().named(EarPlugin.EAR_TASK_NAME).configure(new Action<Task>() {
+                @Override
+                public void execute(Task task) {
+                    sourceSet.getResources().srcDir((Callable) () -> ((Ear) task).getAppDirName());
+                }
+            });
             project.getTasks().withType(Ear.class).configureEach(task -> {
```

Comment:
I guess we don't want to change anything here, but I'm wondering why we configure all the tasks of type Ear here and not the specific task that this plugin registers.

Reply:
We want to set up conventions for all tasks of type `Ear`, even if it defined outside of the plugin (in a build script for instance).